src/chips/sam4l/spi_dma.rs

```diff
@@ -0,0 +1,383 @@
+use helpers::*;
```

Comment:
I think it's useful to keep a completely blocking implementation around, so that you can fall back if needed. You could name them spi and spi_blocking, but meh.